planner/select.go

```diff
@@ -296,20 +296,22 @@ func (n *selectNode) initSource() ([]aggregateNode, error) {
 	return aggregates, nil
 }
 
-func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.FieldDescription] {
+func findFilteredByIndexedField(scanNode *scanNode) immutable.Option[client.IndexDescription] {
 	if scanNode.filter != nil {
```

Comment:
suggestion: Inverting this if statement would make it easier to read IMO.

Reply:
adjusted. Not part of this PR though

Reply:
cheers Islam